Patch-Source: https://github.com/waydroid/waydroid-sensors/pull/9
---
From b5371713d906f85eba8b540afe6d26f732cff4b6 Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Sun, 7 Dec 2025 21:31:20 +0100
Subject: [PATCH] Add systemd system service

Allows installation of a canonical systemd service that starts and supervises
waydroid-sensord.

This file is taken from our downstream services package in postmarketOS
and upstreaming it could make future improvements easier and downstream
packaging.

I'm not the best at CMake configuration, so please let me know if there
is something to nag about.
---
 .gitignore                  |  1 +
 CMakeLists.txt              | 10 ++++++++++
 waydroid-sensord.service.in | 15 +++++++++++++++
 3 files changed, 26 insertions(+)
 create mode 100644 waydroid-sensord.service.in

diff --git a/.gitignore b/.gitignore
index 46f42f8..3bf5b79 100644
--- a/.gitignore
+++ b/.gitignore
@@ -9,3 +9,4 @@ install_manifest.txt
 compile_commands.json
 CTestTestfile.cmake
 _deps
+waydroid-sensord.service
diff --git a/CMakeLists.txt b/CMakeLists.txt
index edd3dbf..5e6243c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,3 +48,13 @@ target_include_directories(waydroid-sensord PUBLIC
 )
 
 install(TARGETS waydroid-sensord RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+configure_file(
+	waydroid-sensord.service.in
+	${CMAKE_CURRENT_BINARY_DIR}/waydroid-sensord.service
+)
+
+install(
+	FILES ${CMAKE_CURRENT_BINARY_DIR}/waydroid-sensord.service
+	DESTINATION /usr/lib/systemd/system/
+)
diff --git a/waydroid-sensord.service.in b/waydroid-sensord.service.in
new file mode 100644
index 0000000..8fe5a9b
--- /dev/null
+++ b/waydroid-sensord.service.in
@@ -0,0 +1,15 @@
+[Unit]
+Description=Waydroid sensor daemon
+Requires=waydroid-container.service
+PartOf=waydroid-container.service
+After=waydroid-container.service
+
+[Service]
+Type=simple
+ExecStart=${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}/waydroid-sensord
+RestartSec=1
+TimeoutStartSec=5
+Restart=always
+
+[Install]
+WantedBy=multi-user.target
